src/core/thread/mle.cpp

```diff
@@ -246,6 +252,7 @@ ThreadError Mle::Stop(void)
         mNetif.RemoveUnicastAddress(mLeaderAloc);
     }
 
+    mNetif.GetPendingDataset().StopDelayTimer();
```

Comment:
Yes, I think that's better.  I think this will bring the code closer to when we perform a full reboot of the node.